runtime/runtime/src/actions.rs

```diff
@@ -686,11 +711,73 @@ pub(crate) fn check_account_existence(
     Ok(())
 }
 
+#[cfg(feature = "protocol_feature_add_account_versions")]
+pub fn recalculate_usage(
+    account: &mut Account,
+    account_id: &AccountId,
+    usage_config: &StorageUsageConfig,
+    state_update: &TrieUpdate,
+) -> Result<(), StorageError> {
+    if let Account::AccountV1(_) = account {
+        // As this is valid account that fits into state we can use unchecked operations here
+        let mut usage: StorageUsage = usage_config.num_bytes_account;
+        if let Some(code) = get_code(state_update, account_id, Some(account.code_hash()))? {
+            usage += code.code.len() as u64;
+        }
+        state_update
+            .iter(&trie_key_parsers::get_raw_prefix_for_access_keys(&account_id))?
+            .map::<Result<(), StorageError>, _>(|raw_key| {
+                let public_key =
+                    trie_key_parsers::parse_public_key_from_access_key_key(&raw_key?, account_id)
+                        .map_err(|_e| {
+                        StorageError::StorageInconsistentState(
+                            "Can't parse public key from raw key for AccessKey".to_string(),
+                        )
+                    })?;
+                usage += usage_config.num_extra_bytes_record
+                    + public_key.try_to_vec().unwrap().len() as u64
+                    + get_access_key(state_update, account_id, &public_key)
+                        .unwrap()
+                        .try_to_vec()
+                        .unwrap()
+                        .len() as u64;
+                Ok(())
+            })
+            .collect::<Result<Vec<_>, _>>()?;
```

Comment:
Why are we collecting to a Vec and throwing the vector itself away? Is this just for side-effects (exhausting the iterator)? In that case, you can use `try_for_each` instead of `map/collect`. 

Useless bit of trivia: `.collect::<Result<(), _>>` would also work. Collecting into `()` works like a for/for_each.

Reply:
That's exactly what I needed, thanks. I guess I just wanted to make that work and then forgot to clean up